mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -155,201 +191,530 @@ public void testSimpleAttachment2() throws MessagingException, IOException {
 
         mailet.service(mail);
 
-        ByteArrayOutputStream rawMessage = new ByteArrayOutputStream();
-        mail.getMessage().writeTo(rawMessage,
-                new String[]{"Bcc", "Content-Length", "Message-ID"});
-        // String res = rawMessage.toString();
-
         @SuppressWarnings("unchecked")
-        Collection<String> c = (Collection<String>) mail
-                .getAttribute(StripAttachment.SAVED_ATTACHMENTS_ATTRIBUTE_KEY);
-        Assert.assertNotNull(c);
-
-        Assert.assertEquals(1, c.size());
+        Collection<String> savedAttachments = (Collection<String>) mail.getAttribute(StripAttachment.SAVED_ATTACHMENTS_ATTRIBUTE_KEY);
+        assertThat(savedAttachments).isNotNull();
+        assertThat(savedAttachments).hasSize(1);
 
-        String name = c.iterator().next();
+        String attachmentFilename = savedAttachments.iterator().next();
 
-        File f = new File("./" + name);
-        try {
-            InputStream is = new FileInputStream(f);
-            String savedFile = toString(is);
-            is.close();
-            Assert.assertEquals(body, savedFile);
-        } finally {
-            FileUtils.deleteQuietly(f);
-        }
+        assertThat(new File(folderPath + attachmentFilename)).hasContent(expectedAttachmentContent);
     }
 
     @Test
-    public void testSimpleAttachment3() throws MessagingException, IOException {
+    public void serviceShouldDecodeFilenameAndSaveAttachmentInAFolderWhenPatternMatchAndDecodeFilenameTrue() throws MessagingException, IOException {
         Mailet mailet = initMailet();
 
-        // System.setProperty("mail.mime.decodefilename", "true");
+        MimeMessage message = mimeMessage();
 
-        MimeMessage message = new MimeMessage(Session
-                .getDefaultInstance(new Properties()));
-
-        MimeMultipart mm = new MimeMultipart();
-        MimeBodyPart mp = new MimeBodyPart();
-        mp.setText("simple text");
-        mm.addBodyPart(mp);
-        String body = "\u0023\u00A4\u00E3\u00E0\u00E9";
-        MimeBodyPart mp2 = new MimeBodyPart(new ByteArrayInputStream(
-                ("Content-Transfer-Encoding: 8bit\r\n\r\n" + body).getBytes("UTF-8")));
-        mp2.setDisposition("attachment");
-        mp2
-                .setFileName("=?iso-8859-15?Q?=E9_++++Pubblicit=E0_=E9_vietata____Milano9052.tmp?=");
-        mm.addBodyPart(mp2);
-        String body2 = "\u0014\u00A3\u00E1\u00E2\u00E4";
-        MimeBodyPart mp3 = new MimeBodyPart(new ByteArrayInputStream(
-                ("Content-Transfer-Encoding: 8bit\r\n\r\n" + body2).getBytes("UTF-8")));
-        mp3.setDisposition("attachment");
-        mp3.setFileName("temp.zip");
-        mm.addBodyPart(mp3);
+        MimeMultipart multiPart = new MimeMultipart();
+        MimeBodyPart part = new MimeBodyPart();
+        part.setText("simple text");
+        multiPart.addBodyPart(part);
+        String expectedAttachmentContent = "\u0023\u00A4\u00E3\u00E0\u00E9";
```

Comment:
benoit suggested to put this repeated string into a constant, I think it could be a good idea.